lib/pbench/test/unit/server/test_datasets_list.py

```diff
@@ -698,3 +698,71 @@ def test_key_summary(self, query_as):
                 "origin": None,
             },
         }
+
+    def get_daterange_results(
+        self, name_list: list[str]
+    ) -> dict[str, datetime.datetime]:
+        """
+        Use a list of "expected results" to determine the earliest and the
+        latest creation date of the set of datasets.
+
+        Args:
+            name_list: List of dataset names
+
+        Returns:
+            {"from": first_date, "to": last_date}
+        """
+        from_time = datetime.datetime.now(datetime.timezone.utc)
+        to_time = datetime.datetime(
+            year=1970, month=1, day=1, tzinfo=datetime.timezone.utc
+        )
+        for name in sorted(name_list):
+            dataset = Dataset.query(name=name)
+            to_time = max(dataset.uploaded, to_time)
+            from_time = min(dataset.uploaded, from_time)
```

Comment:
Alternatively, you could do something like this:
```python
        to_time = max(Dataset.query(name=n).uploaded for n in name_list)
        from_time = min(Dataset.query(name=n).uploaded for n in name_list)
```
This does do two database queries, but in the context of the unit tests that's probably not an issue.  On the other hand, it saves you from having to initialize `to_time` and `from_time` to crazy values.  (BTW, I don't see any reason why we need to sort `name_list` before aggregating it...do you?)

However, it has a nasty failure mode when `name_list` is empty...but, then, your proposed code has an "interesting" behavior in that case, too....  🙂

Reply:
And, um, it turns out that the query itself has a problem in that case, so I *do* need to deal with the empty list case, on both ends. And the `sorted` is, yeah, unnecessary here, but was simply copied originally from the datasets list test and apparently didn't seem worth changing. Huh.